x-pack/solutions/security/plugins/security_solution/common/siem_migrations/resources/resource_identifier_base.ts

```diff
@@ -69,25 +73,33 @@ export abstract class ResourceIdentifier<I> {
     ];
   }
 
-  public fromResource(resource: SiemMigrationResourceData): SiemMigrationResourceBase[] {
-    if (resource.type === 'macro' && resource.content) {
+  public async fromResource(
+    resource: SiemMigrationResourceData
+  ): Promise<SiemMigrationResourceBase[]> {
+    if (resource.type === 'macro' && resource.content && this.vendor === 'splunk') {
       return this.identifier(resource.content);
     }
-    return [];
+    return Promise.resolve([]);
   }
 
-  public fromResources(resources: SiemMigrationResourceData[]): SiemMigrationResourceBase[] {
+  public async fromResources(
+    resources: SiemMigrationResourceData[]
+  ): Promise<SiemMigrationResourceBase[]> {
+    if (this.vendor !== 'splunk') {
+      return [];
+    }
     const lookups = new Set<string>();
     const macros = new Set<string>();
-    resources.forEach((resource) => {
-      this.fromResource(resource).forEach((identifiedResource) => {
+    for (const resource of resources) {
+      const resourcesFromResource = await this.fromResource(resource);
```

Comment:
probably can move this await out of this loop as well

Reply:
Yes.. this file does not have tests so i will rather not change it for now. I will add tests later and then i will confidence changing it.